### ISSUE TITLE
Cannot set a custom java.time.LocalTime encoder

### ISSUE THREAD

**Issue by vemv** [metosin/jsonista#33]
Hi!

It appears that if passing a custom java.time.LocalTime encoder, it will not work at all.

As a proof, only one of the following two snippets `println`s something:

```clj
(jsonista/write-value-as-string (java.time.LocalTime/now)
                                (jsonista/object-mapper {:encoders
                                                         {java.time.LocalTime (fn [x jg]
                                                                                (println "Do I work?")
                                                                                (.writeString jg (.toString x)))}}))

(jsonista/write-value-as-string (java.util.Date.)
                                (jsonista/object-mapper {:encoders
                                                         {java.util.Date (fn [x jg]
                                                                           (println "I work!")
                                                                           (.writeString jg (.toString x)))}}))
```

The following seems related:

https://github.com/metosin/jsonista/blob/211306f04bb15d7232b536cf6c6d8ecfeae0512d/src/clj/jsonista/core.clj#L70

Thanks  - V

**Comment by vemv** [comment]
I could fix this issue by swapping the order of these two lines:

https://github.com/metosin/jsonista/blob/211306f04bb15d7232b536cf6c6d8ecfeae0512d/src/clj/jsonista/core.clj#L130-L131

I verified that both my project's bugrepro would be fixed, and jsonista's test suite would keep passing.

Cannot open a PR myself sadly. Would be very thankful for a fix!

**Comment by ikitommi** [comment]
shipped with `[metosin/jsonista "0.2.7"]`
